cartographer/mapping/map_builder.cc

```diff
@@ -361,20 +365,6 @@ std::map<int, int> MapBuilder::LoadState(
     }
   }
 
-  // TODO(schwoere): Remove backwards compatibility once the pbstream format
-  // version 2 is established.
-  if (deserializer.header().format_version() ==
-      io::kFormatVersionWithoutSubmapHistograms) {
-    submap_id_to_submap =
-        cartographer::io::MigrateSubmapFormatVersion1ToVersion2(
```

Comment:
Do we still need this conversion in the library? I also created issue #1709 regarding this.